### PR TITLE
#248 - Compatability of examples with new ui module

### PR DESCRIPTION
**Pull request by abhishek1508**
Fix example to be compatible with new `libnavigation-ui` module [#248](https://github.com/mapbox/navigation-sdks/issues/248)

- [x] I have added any issue links
- [x] I have added all related labels (`bug`, `feature`, `new API(s)`, `SEMVER`, etc.)
- [x] I have added the appropriate milestone and project boards

### Goal

Fix example to be compatible with new `libnavigation-ui` module

### Implementation

Fix example to be compatible with new `libnavigation-ui` module

## Testing

Please describe the manual tests that you ran to verify your changes

- [x] I have tested locally (including `SNAPSHOT` upstream dependencies if needed) through testapp/demo app and run all activities to avoid regressions
- [x] I have tested via a test drive, or a simulation/mock location app
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have updated the `CHANGELOG` including this PR

### CODE REVIEW

File: examples/src/main/java/com/mapbox/navigation/examples/settings/OfflineRegionDownloadActivity.kt [hunk 1]
```diff
@@ -32,16 +31,11 @@ import com.mapbox.mapboxsdk.style.sources.GeoJsonSource
...
-import com.mapbox.services.android.navigation.v5.navigation.OfflineTiles
-import com.mapbox.services.android.navigation.v5.navigation.OnOfflineTilesRemovedCallback
-import com.mapbox.services.android.navigation.v5.navigation.OnTileVersionsFoundCallback
-import com.mapbox.services.android.navigation.v5.navigation.RouteTileDownloadListener
+import com.mapbox.navigation.route.onboard.OnOfflineTilesRemovedCallback
 import kotlinx.android.synthetic.main.activity_offline_region_download.*
 import org.json.JSONObject
 
-class OfflineRegionDownloadActivity : AppCompatActivity(), RouteTileDownloadListener,
+class OfflineRegionDownloadActivity : AppCompatActivity(), /*RouteTileDownloadListener,*/
```
Comment (RingerJK) [inline examples/src/main/java/com/mapbox/navigation/examples/settings/OfflineRegionDownloadActivity.kt]: don't have opportunity to download tiles now. https://github.com/mapbox/mapbox-navigation-android/issues/2493 . Suggest to remove this activity

File: libnavigation-ui/build.gradle [hunk 2]
```diff
@@ -63,8 +63,8 @@ dependencies {
   implementation project(':libnavigation-util')
 
   // Mapbox
+  api dependenciesList.mapboxSdkTurf
```
Comment (RingerJK) [inline libnavigation-ui/build.gradle]: why did `Turf` dependency stay `api`?

Comment (abhishek1508) [reply libnavigation-ui/build.gradle]: Examples is using `Turf` related constants. Hence it was changed from `implementation` to `api`

File: examples/src/main/java/com/mapbox/navigation/examples/settings/OfflineRegionDownloadActivity.kt (hunk 1 above)
Comment (abhishek1508) [reply examples/src/main/java/com/mapbox/navigation/examples/settings/OfflineRegionDownloadActivity.kt]: Yeah will remove this activity. I will also be removing `MockNavigationActivity`

File: libnavigation-ui/build.gradle (hunk 2 above)
Comment (Guardiola31337) [reply libnavigation-ui/build.gradle]: In the legacy, we're exposing `Turf` in `libandroid-navigation` https://github.com/mapbox/mapbox-navigation-android/blob/fa2e50c2661bfbfacdad14bd8f3c967237beb34f/libandroid-navigation/build.gradle#L60 and not anymore 👀 https://github.com/mapbox/mapbox-navigation-android/blob/fa2e50c2661bfbfacdad14bd8f3c967237beb34f/libnavigation-core/build.gradle#L40 Do we want to do the same so this dependency wouldn't be needed in `libnavigation-ui` at all?

Comment (abhishek1508) [reply libnavigation-ui/build.gradle]: Yeah we can change the `implementation` to `api` in `libnavigation-core/build.gradle`

**Comment by abhishek1508** [comment]
`OfflineRegionDownloadActivity` and `MockNavigationActivity` would come in  eventually. Will cut tickets for these.
We will clean up `strings.xml` in further iterations.

**Pull request merged**